### PR TITLE
Fix/provider archive memory spike

### PR DESCRIPTION
**Pull request by brooksmtownsend**
@thomastaylor312 wrote the vast majority of this PR, I only stepped in to deal with inconsistent tests.

This PR upgrades the wasmcloud NIF to use a newer version of `provider-archive` which more efficiently unpacks provider archives for memory constrained devices.

The TL;DR regarding inconsistent tests is that there was a race condition between a provider receiving hostdata with link definitions and actually subscribing to the linkdef topic so we were intermittently losing link definitions, causing HTTP requests and keyvalue invocations to fail

### CODE REVIEW

**Review by autodidaddict — APPROVED** [review]

**Comment by thomastaylor312** [comment]
Ok, this has been updated with the latest provider-archive version and should be ready for final review

**Review by autodidaddict — APPROVED** [review]

File: host_core/native/hostcore_wasmcloud_native/src/par.rs [hunk 1]
```diff
@@ -40,18 +40,18 @@ pub(crate) fn extract_target_bytes(par: &ProviderArchive) -> Result<Vec<u8>, Err
 }
 
 pub(crate) fn cache_path(
-    subject: String,
-    rev: u32,
-    contract_id: String,
-    link_name: String,
+    subject: &str,
+    rev: i32,
```
Comment (brooksmtownsend) [inline host_core/native/hostcore_wasmcloud_native/src/par.rs:44]: I think this should still be `u32` unless there's a reason I'm missing to support negative numbers (feels weird for filepath)

Comment (autodidaddict) [reply host_core/native/hostcore_wasmcloud_native/src/par.rs:44]: yeah revisions are never negative.

Comment (thomastaylor312) [reply host_core/native/hostcore_wasmcloud_native/src/par.rs:44]: I had actually changed it, but the actual claims type from wascap is an `i32`, so it didn't really make sense to change it. I just changed it to match what things were set to

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**